packages/components/src/dropdown-menu/v2/stories/index.tsx

```diff
@@ -0,0 +1,264 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+import styled from '@emotion/styled';
+
+/**
+ * Internal dependencies
+ */
+import { COLORS } from '../../../utils';
+import {
+	DropdownMenu,
+	DropdownMenuItem,
+	DropdownSubMenu,
+	DropdownMenuSeparator,
+	DropdownMenuCheckboxItem,
+	DropdownMenuGroup,
+	DropdownMenuLabel,
+	DropdownMenuRadioGroup,
+	DropdownMenuRadioItem,
+	DropdownSubMenuTrigger,
+} from '..';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, createContext, useContext } from '@wordpress/element';
+import { menu, wordpress } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Icon from '../../../icon';
+
+const DropdownMenuStoryContext = createContext< {
+	itemOneChecked?: boolean;
+	setItemOneChecked?: ( v: boolean ) => void;
+	itemTwoChecked?: boolean;
+	setItemTwoChecked?: ( v: boolean ) => void;
+	radioValue?: string;
+	setRadioValue?: ( v: string ) => void;
+} >( {} );
+
+const meta: ComponentMeta< typeof DropdownMenu > = {
+	title: 'Components (Experimental)/DropdownMenu v2',
+	component: DropdownMenu,
+	subcomponents: {
+		DropdownMenuItem,
+		DropdownSubMenu,
+		DropdownSubMenuTrigger,
+		DropdownMenuSeparator,
+		DropdownMenuCheckboxItem,
+		DropdownMenuGroup,
+		DropdownMenuLabel,
+		DropdownMenuRadioGroup,
+		DropdownMenuRadioItem,
+	},
+	argTypes: {
+		children: { control: { type: null } },
+		trigger: { control: { type: null } },
+	},
+	parameters: {
+		actions: { argTypesRegex: '^on.*' },
+		controls: { expanded: true },
+		docs: { source: { state: 'open', excludeDecorators: true } },
+	},
+	decorators: [
```

Comment:
Just out of curiosity so I fully understand, could you briefly explain:

- Why are we using decorators instead of putting them directly in the `Template`? (Is it so we can benefit from `excludeDecorators`?)
- Why are we using `context`?

Reply:
Good point. No action needed now after we followed your [suggestion](https://github.com/WordPress/gutenberg/pull/49473#discussion_r1198784346) and we kept the original implementation in the same folder

Reply:
Re. the wrapper, the reason why I'm using a decorator + `excludeDecorators` is that is keeps the code snippet shown in the docs tab cleaner IMO